diff -urN a/servers/linux-outloud/Makefile b/servers/linux-outloud/Makefile 
--- a/servers/linux-outloud/Makefile	2016-04-21 02:54:59.000000000 +0200
+++ b/servers/linux-outloud/Makefile	2016-11-20 11:52:34.115201000 +0100
@@ -5,7 +5,7 @@
 LIBPARENTDIR = ${PREFIX}/share/emacs/site-lisp
 LIBDIR =$(LIBPARENTDIR)/emacspeak/servers/linux-outloud
 
-TCL_VERSION=8.6
+TCL_VERSION=8.4
 TCL_INCLUDE=/usr/include/tcl$(TCL_VERSION)
 #CFLAGS+= -g    -O2 -fPIC  -DPIC  -pedantic -ansi -Wall  -Wno-long-long -I$(TCL_INCLUDE)
 CFLAGS+= -m32 -g    -O2 -fPIC  -DPIC  -pedantic -ansi -Wall  -Wno-long-long -I$(TCL_INCLUDE)
